src/Avalonia.Controls/Converters/CornerRadiusFilterConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿#nullable enable
+using System;
+using System.Globalization;
+
+using Avalonia.Data.Converters;
+
+namespace Avalonia.Controls.Converters
+{
+    /// <summary>
+    /// Converts an existing CornerRadius struct to a new CornerRadius struct,
+    /// with filters applied to extract only the specified fields, leaving the others set to 0.
+    /// </summary>
+    public class CornerRadiusFilterConverter : IValueConverter
+    {
+        /// <summary>
+        /// Gets or sets the type of the filter applied to the <see cref="CornerRadiusFilterConverter"/>.
+        /// </summary>
+        public CornerRadiusFilterKinds Filter { get; set; }
```

Comment:
I'm wondering - do we need to make this a direct property to support basic bindings? Or we are fine with having a simpler version?

Reply:
I am not sure, how it's useful to have bindings with resources, where this converter will live most of the time.
Although it should be possible to create its instance inside of Binding with specifying another binding for this property. Don't know if it's needed for anybody though.

Reply:
I guess if we decide to add it later it won't be a huge breaking change.

Reply:
But anyway - first class compiler support for such constructs will solve this. Since in my codebase I have something similar for `Thickness` as well. And I guess many people have converters that just process one object into another variant.